src/manifest.ts

```diff
@@ -1114,28 +1132,131 @@ export class Manifest {
       }
     }
 
-    if (this.sequentialCalls) {
-      const resultReleases: CreatedRelease[] = [];
-      for (const pullNumber in releasesByPullRequest) {
-        const releases = await this.createReleasesForPullRequest(
-          releasesByPullRequest[pullNumber],
-          pullRequestsByNumber[pullNumber]
-        );
-        resultReleases.push(...releases);
-      }
-      return resultReleases;
-    } else {
-      const promises: Promise<CreatedRelease[]>[] = [];
-      for (const pullNumber in releasesByPullRequest) {
-        promises.push(
-          this.createReleasesForPullRequest(
+    // to minimize risks of race condition we lock branches used as the source of commits for the duration of the
```

Comment:
Not for now, but would renaming the branch to something temporary before releasing avoid the racing condition altogether?

Reply:
Hmm, I don't believe that would avoid the race condition. I think that would be a different approach to achieving a branch lock, but with the same base risks, and few others specific to the renaming.

Race condition risks when locking branch via permission rules:
- commits may have been sent to `next` between the release PR merge and the time we set the lock

Race condition risks when locking branch by renaming it:
- commits may have been sent to `new` between the release PR merge and the time we rename the branch
- the branch may have been recreated (by someone/something pushing their local `next` branch), which would result in a conflict when trying to rename the "locked" branch back to `next`
- any permission rules, ruleset, or other meta config targeting `next` would not be applied during the time the branch is renamed

That's theoretical, I'm not sure if that would be too much of a problem in practice, but renaming the branch has a wider scope for conflicts IMHO.

Reply:
To be honest I don't really like relying on permission rules to lock the branch, I would be happy to find another solution.

I'm also considering that we could not lock at all, and instead check twice for race conditions.

`createReleases()` would do the following:

1. find all release pull requests merged since last release
2. `throwIfChangesBranchesRaceConditionDetected()`: check for any race condition for found pull request "changes branches"
3. `createReleasesForPullRequest()` (done once per release pull request found)
4. `throwIfChangesBranchesRaceConditionDetected()`: creating releases may have taken some time, check again for race conditions before aligning the branch to be sure we don't overwrite new commits
5. `alignPullRequestsChangesBranch()`

There is still a risk of race condition between step 4 and 5, so it's not a safe as the branch lock approach, but the risk window is fairly small and we wouldn't need to touch branch permission rules.